From 12c448214f203ab1773f914970c06aa07ba529cf Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Fri, 13 Apr 2018 17:23:35 +0800
Subject: [PATCH 3703/5242] MLK-17631-6 usb: cdns3: gadget: disable U1 and U2
 for reset state

commit  37e1a8b1f7a53fba1c98f7817e8fa91cc09bcf1c from
https://source.codeaurora.org/external/imx/linux-imx.git

When we receive the bus reset, according to CH 9.4.5, the U1
and U2 should be reset to disabled status.

Reviewed-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index 52375a9..7025f1f 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -206,7 +206,7 @@ static void cdns_ep_stall_flush(struct usb_ss_endpoint *usb_ss_ep)
  */
 static void cdns_ep0_config(struct usb_ss_dev *usb_ss)
 {
-	u32 max_packet_size = 0;
+	u32 reg, max_packet_size = 0;
 
 	switch (usb_ss->gadget.speed) {
 	case USB_SPEED_UNKNOWN:
@@ -274,6 +274,10 @@ static void cdns_ep0_config(struct usb_ss_dev *usb_ss)
 		EP_STS_EN__SETUPEN__MASK |
 		EP_STS_EN__TRBERREN__MASK);
 
+	reg = gadget_readl(usb_ss, &usb_ss->regs->usb_conf);
+	reg |= USB_CONF__U1DS__MASK | USB_CONF__U2DS__MASK;
+	gadget_writel(usb_ss, &usb_ss->regs->usb_conf, reg);
+
 	cdns_prepare_setup_packet(usb_ss);
 }
 
-- 
1.7.9.5

